packages/go_router/lib/src/delegate.dart

```diff
@@ -185,3 +179,20 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     return SynchronousFuture<void>(null);
   }
 }
+
+/// The route match that represent route pushed through [GoRouter.push].
+// TODO(chunhtai): Removes this once imperative API no longer insert route match.
+class ImperativeRouteMatch extends RouteMatch {
```

Comment:
This is a workaround that should be removed once the imperative API refactor is done